### PR TITLE
Validate chat messages

### PR DESCRIPTION
**Pull request by AnandAili**

### CODE REVIEW

File: src/main/java/chatbox/ChatMessage.java [hunk 1]
```diff
@@ -28,7 +30,9 @@
 
     private final Date creationTime = new Date();
 
-    @NotBlank
+    @NotBlank(message = "Message must have at least " + SmokefreeConstants.ChatBox.MINMUM_MESSAGE_LENGTH + " characters")
+    @Pattern(regexp = "^[A-Za-z0-9\\n!@&(),.?\": ]+$", message = "Please enter only allowed special charaxters: @&(),.?\": ")
```
Comment (dimitriirtimid) [inline src/main/java/chatbox/ChatMessage.java]: need to check what characters we should allow

File: src/main/java/chatbox/ChatMessage.java [hunk 2]
```diff
@@ -28,7 +30,9 @@
 
     private final Date creationTime = new Date();
 
-    @NotBlank
+    @NotBlank(message = "Message must have at least " + SmokefreeConstants.ChatBox.MINMUM_MESSAGE_LENGTH + " characters")
```
[the remark this answers is not in the document]
Comment (AnandAili) [reply src/main/java/chatbox/ChatMessage.java]: Fixed typo mistake

File: src/main/java/chatbox/ChatboxController.java [hunk 3]
```diff
@@ -32,7 +35,7 @@
 
 
     @Post("/{chatboxId}")
-    public ChatMessage postMessage(Authentication authentication, String chatboxId, @Size(max=4096) @Body ChatMessage chatMessage) {
+    public ChatMessage postMessage(Authentication authentication, String chatboxId, @Size(max=4096) @Valid @Body ChatMessage chatMessage) {
```
[the remark this answers is not in the document]
Comment (AnandAili) [reply src/main/java/chatbox/ChatboxController.java]: Yeah, as discussed, removed the size validation here

File: src/main/java/smokefree/aws/rds/secretmanager/SmokefreeConstants.java [hunk 4]
```diff
@@ -29,12 +29,17 @@
 		public static final String EMAIL_ADDRESS = "email";
 	}
 
-	public static final Integer MAXIMUM_PLAYGROUNDS_DISTANCE = 100;
-	public static final Integer MAXIMUM_PLAYGROUNDS_ALLOWED = 1000;
+	public static final int MAXIMUM_PLAYGROUNDS_DISTANCE = 100;
+	public static final int MAXIMUM_PLAYGROUNDS_ALLOWED = 1000;
```
[the remark this answers is not in the document]
Comment (AnandAili) [reply src/main/java/smokefree/aws/rds/secretmanager/SmokefreeConstants.java]: As discussed, renamed the constants

File: src/main/java/chatbox/ChatMessage.java (hunk 1 above)
Comment (dimitriirtimid) [reply src/main/java/chatbox/ChatMessage.java]: drop the character whitelist. We will rely on the front-end to escape.
Please do not yet merge this pr

**Comment by AnandAili** [comment]
closing this PR. created a new PR after java-refactor major changes

**Pull request closed without merging**